examples/EvaluatingChatAssistant/EvaluatingChatAssistant.ipynb

```diff
@@ -0,0 +1,353 @@
+    "    input: \"who won the men's trophy that year?\",\n",
+    "    expected: \"The men's Ballon d'Or trophy in 2018 was won by Luka Modrić of Croatia, who played for Real Madrid.\"\n",
+    "  },\n",
+    "  // ... rest of dataset\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running evals\n",
+    "\n",
+    "The key to running evals on a multi-turn conversation is to include the history of the chat in the chat completion request.\n",
+    "\n",
+    "To do so, we simply add the history between the `system` message and the final `user` message in the `messages` argument as seen below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { wrapOpenAI } from \"braintrust\";\n",
+    "import { OpenAI } from \"openai\";\n",
+    "\n",
+    "async function runTask({\n",
+    "  input,\n",
+    "  chat_history\n",
+    "}: {\n",
+    "  input: string,\n",
+    "  chat_history: {\n",
+    "    role: \"user\" | \"assistant\";\n",
+    "    content: string;\n",
+    "  }[]\n",
+    "}) {\n",
+    "  const client = wrapOpenAI(\n",
+    "    new OpenAI({\n",
+    "      baseURL: \"https://braintrustproxy.com/v1\",\n",
+    "      apiKey: process.env.OPENAI_API_KEY ?? \"\", // Can use OpenAI, Anthropic, Mistral etc. API keys here\n",
+    "    })\n",
+    "  )\n",
+    "\n",
+    "  const response = await client.chat.completions.create({\n",
+    "    model: \"gpt-4o\",\n",
+    "    messages: [\n",
+    "      {\n",
+    "        role: \"system\",\n",
+    "        content: \"You are a helpful and polite assistant who knows about sports.\"\n",
+    "      },\n",
+    "      ...chat_history,\n",
+    "      {\n",
+    "        role: \"user\",\n",
+    "        content: input,\n",
+    "      }\n",
+    "    ]\n",
+    "  });\n",
+    "  return response.choices[0].message.content || \"\";\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Scoring the eval\n",
+    "\n",
+    "We'll use a `Factuality` scoring function to check how the output of the completion compares to the expected value.\n",
+    "\n",
+    "For this cookbook, we will alter the [built in factuality prompt](https://github.com/braintrustdata/autoevals/blob/main/templates/factuality.yaml) slightly.\n",
+    "\n",
+    "(I have used the built-in Factuality scorer and it does not significantly reduce the score compared to running the eval with no chat history. If the model does not have enough context to factually answer the question and responds with a request for the user to be more specific, the Factuality scorer evaluates this response as having no factual inconsistencies between the expert and submitted answers. Because this cookbook is to illustrate how to incorporate chat history to produce a good response, we will tweak the built-in prompt slightly to account for this.)\n",
```

Comment:
Rather than saying the built-in spec doesn't work, I think we can say something along the lines of:

> We will extend the Factuality prompt that that is available in the autoevals open source library to specifically score answers that ask for more context. This will allow our scorer to take advantage of the additional chat history that we will be using blah blah

I think in general it's a good cookbook exercise to describe how to go about creating your own scorer based on the autoeval templates in the open source repo as well, so I think framing it that way might be more helpful

Reply:
Also will be helpful to do something like printing it out in a code block so they can see the full spec inline:

```
import spec from "./assets/factuality"
console.log(spec);
```

Reply:
It could also be useful to run the default `Factuality`, show the experiment results, and then how you arrived at the conclusion that it'd benefit from some improvement. And make that the next experiment.